test/runtests_parser.py

```diff
@@ -1,293 +0,0 @@
-import json
-from os import path
-
-
-class Test:
-    output = {}
-    unique_id = set()
-
-    @staticmethod
-    def register_output(idnum, filename, overwrite=True):
-        if not overwrite and Test.has_output(filename):
-            return
-
-        Test.output[filename] = idnum
-
-    @staticmethod
-    def has_output(filename):
-        return filename in Test.output
-
-    @staticmethod
-    def filename_to_test_id(filename):
-        return Test.output[filename]
-
-    @staticmethod
-    def check_add_unique_id(idnum):
-        if idnum in Test.unique_id:
-            raise Exception("test id is repeated. fatal id: " + str(idnum))
-        else:
-            Test.unique_id.add(idnum)
-
-    def __init__(self, id, desc):
-        self.id = id
-        self.desc = desc.strip()
-        self.backslash_seen = False
-        self.vw_command = ""
-        self.is_bash_command = False
-        self.diff_files = []
-
-    def add_more_comments(self, line):
-        self.desc = self.desc + ". " + line.strip()[1:].strip()
-
-    def add_vw_command(self, line):
-        self.vw_command = (self.vw_command + " " + line).strip()
-        if self.vw_command[-1] == "\\":
-            self.vw_command = self.vw_command[:-1]
-            self.backslash_seen = True
-
-        if "&&" in self.vw_command:
-            self.is_bash_command = True
-
-    # a cmd is incomplete if self.backslash_seen is true
-    # this represents that the previous command parsed ended in '\' char
-    def append_cmd_if_incomplete(self, line):
-        if self.backslash_seen:
-            self.backslash_seen = False
-            self.add_vw_command(line)
-            return True
-        else:
-            return False
-
-    def add_file(self, line):
-        self.diff_files.append(line)
-
-    def add_bash_command(self, line):
-        if self.vw_command:
-            raise Exception(
-                "construction faulty merging" + self.vw_command + " and " + line
-            )
-        self.is_bash_command = True
-        self.add_vw_command(line)
-
-    def clean(self):
-        if not self.is_bash_command:
-            self.vw_command = " ".join(self.vw_command.split()[1:])
-
-            # check what input files this command needs
-            input_file_flags = [
```

Comment:
is this processing done elsewhere now? this and checking for missing and repeat id numbers?

Reply:
Input files are up to the test author. I think checking for missing and duplicate ids is missing, I'll add that in

Reply:
input files and test dependency could still be aided by a script, or at least as a verification step

Reply:
You can't depend on it for bash_command tests. I guess you could make it automatic for vw_command tests and explicit for bash_command tests. I initially didn't do that as it felt inconsistent. It would reduce effort in adding new tests though.